Automattic/woocommerce-payments
# Add onboarding page for progressive onboarding prototype

Fixes #4811

#### Changes proposed in this Pull Request
- Rename the old KYC experiment files, to avoid any conflict with the prototype and project. We might end using some code from it.
- Add the new onboarding prototype page and ensure the merchant gets redirected there instead of Stripe KYC.
- Add a notice and a button to allow using the regular onboarding process.
- Implement `fromDotNotation` util to unflatten objects. It's handy to have single level objects as state using dot notation and expand them before they're sent to the server.
- Initial implementation of the form, only for `business_type: individual` for now. Using URL query to send the form data to the PHP side. Sanitizing and formatting before it's being sent to the server according to [the current server draft PR](2511-gh-Automattic/woocommerce-payments-server).

#### Testing instructions
- Checkout the server branch  `dev/2469-support-creation-of-progressive-onboarding-acct`
- Without an onboarded account proceed with the onboarding as usual.
- Whatever method you use to onboard (task, Payments > Overview...) should land on the new Onboarding page: `admin.php?page=wc-admin&path=/payments/onboarding-prototype`
- Clicking on **Connect** on the top notice should allow you to onboard as usual.
- Fill the form and click on **Submit**
- You should be redirected to Stripe with all the details prefilled and progressive onboarding enabled.

#### Screenshots
<img width="1447" alt="Screenshot 2022-09-30 at 13 06 21" src="https://user-images.githubusercontent.com/7670276/193256807-0dced77b-4238-431e-965f-dc74c2b50a23.png">

-------------------


**Post merge**
